index.js

```diff
@@ -1,95 +1,138 @@
-// console.log("%c----------", "color: red") 
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red")
+
+function sameSameButDifferent(num, maybeNum) {
+  if (num === maybeNum) {
+    return "same same"
+  } else if (num === parseInt(maybeNum)) {
+    return "same same (but different)"
+  } else {
+    return "different"
+  }
+}
+
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
+
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
-//
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
-//
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
-//
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
+
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student.grade = grade
+}
 
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)  
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red") 
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)  
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 6 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+console.log("%cQuestion 6", "color: red")
+
+// function printNameAndPhones(users) {
+//   for (i = 0; i < users.length; i++) {
+//     console.log(users[i].name)
+//     console.log(`Cell: ${users[i].phones.cell}`)
+//     console.log(`Office: ${users[i].phones.office}`)
 //   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
-// // => "Cell: 555-123-4567"
-// // => "Office: 555-456-7890"
-// // => "Liza"
-// // => "Cell: 555-234-5678"
-// // => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
+// }
+
+function showUserInfo(userObj) {
+  console.log(userObj.name)
+  console.log(`Cell: ${userObj.name}`)
+  console.log(`Office: ${userObj.name}`)
+}
+ //callback function used
+function printNameAndPhones(users) {
+  users.forEach(showUserInfo)
+}
```

Comment:
Nice use of the callback function here! If you get used to this syntax, and the concept of passing around function references, you'll be in great shape for learning JS.

Reply:
I refactored this after the walkthrough. I actually feel better about doing a callback instead of the nested function. Still getting up to speed on callbacks but otherwise feel good about JavaScript.